spec.html

```diff
@@ -33849,161 +33849,97 @@ <h1>String.prototype.replace ( _searchValue_, _replaceValue_ )</h1>
           1. Else,
             1. Assert: Type(_replaceValue_) is String.
             1. Let _captures_ be a new empty List.
             1. Let _replacement_ be ! GetSubstitution(_searchString_, _string_, _position_, _captures_, *undefined*, _replaceValue_).
-          1. Return the string-concatenation of _preserved_, _replacement_, and the substring of _string_ from _position_ + _searchLength_.
+          1. Return the string-concatenation of _preceding_, _replacement_, and _following_.
         </emu-alg>
         <emu-note>
           <p>The `replace` function is intentionally generic; it does not require that its *this* value be a String object. Therefore, it can be transferred to other kinds of objects for use as a method.</p>
         </emu-note>
 
-        <emu-clause id="sec-getsubstitution" type="abstract operation">
+        <emu-clause id="sec-getsubstitution" type="abstract operation" oldids="table-replacement-text-symbol-substitutions,table-45">
           <h1>
             GetSubstitution (
               _matched_: a String,
               _str_: a String,
               _position_: a non-negative integer,
-              _captures_: a possibly empty List of Strings,
+              _captures_: a possibly empty List, each of whose elements is a String or *undefined*,
               _namedCaptures_: an Object or *undefined*,
-              _replacement_: a String,
+              _replacementTemplate_: a String,
             )
           </h1>
           <dl class="header">
+            <dt>description</dt>
+            <dd>For the purposes of this abstract operation, a <em>decimal digit</em> is a code unit in the range 0x0030 (DIGIT ZERO) to 0x0039 (DIGIT NINE) inclusive.</dd>
           </dl>
           <emu-alg>
-            1. Let _matchLength_ be the number of code units in _matched_.
             1. Let _stringLength_ be the number of code units in _str_.
             1. Assert: _position_ &le; _stringLength_.
-            1. Let _tailPos_ be _position_ + _matchLength_.
-            1. Let _m_ be the number of elements in _captures_.
-            1. Let _result_ be the String value derived from _replacement_ by copying code unit elements from _replacement_ to _result_ while performing replacements as specified in <emu-xref href="#table-replacement-text-symbol-substitutions"></emu-xref>. These `$` replacements are done left-to-right, and, once such a replacement is performed, the new replacement text is not subject to further replacements.
+            1. Let _templateRemainder_ be _replacementTemplate_.
+            1. Let _result_ be the empty String.
+            1. Repeat, while _templateRemainder_ is not the empty String,
+              1. NOTE: The following steps will isolate _ref_ (a prefix of _replacementTemplate_), determine _refReplacement_ (its replacement), and then add that to _result_.
```

Comment:
```suggestion
              1. NOTE: The following steps isolate _ref_ (a prefix of _replacementTemplate_), determine _refReplacement_ (its replacement), and then append that replacement to _result_.
```

Reply:
Done. Plus I noticed that `_replacementTemplate_` should be `_templateRemainder_` there.